pyvo/dal/adhoc.py

```diff
@@ -169,6 +169,109 @@ class DatalinkResultsMixin(AdhocServiceResultsMixin):
     """
     Mixin for datalink functionallity for results classes.
     """
+    def iter_datalinks_from_dlblock(self, datalink_service):
```

Comment:
I would think both of these new methods could be prepended with `_`, unless you envision them to be used by the end users.

Reply:
Agree. While I see the merits of invoking only this method (the other one seems more adhoc - hence the best efforts that are needed here), for the regular user it's just an implementation detail and it shouldn't crop up into the API.